File: PayTR Direkt API/PayTR BIN Sorgulama Servisi/English/Sample PYTHON/getBin.py
# Python 3.6+


import base64
import hmac
import hashlib
import requests
import json


merchant_id = 'XXX'
merchant_key = 'XXX'
merchant_salt = 'XXX'


bin_number = 'XXXXXX'


hash_str = bin_number + merchant_id + merchant_salt
paytr_token = base64.b64encode(hmac.new(merchant_key, hash_str.encode(), hashlib.sha256).digest())

params = {
    'merchant_id': merchant_id,
    'bin_number': bin_number,
    'paytr_token': paytr_token
}

result = requests.post('https://www.paytr.com/odeme/api/bin-detail', params)
res = json.loads(result.text)

if res['status'] == 'error':
    print('PAYTR BIN detail request error. Error: ' + res['err_msg'])
elif res['status'] == 'failed':
    print('BIN tanımlı değil. (Örneğin bir yurtdışı kartı)')
else:
    print(result.text)

File: PayTR Direkt API/PayTR BIN Sorgulama Servisi/English/Example PHP/getBin.php
<?php

	$merchant_id 	= 'XXXXXX';
	$merchant_key 	= 'XXXXXX';
	$merchant_salt	= 'XXXXXX';
	#
	$bin_number	= "";
	#
	############################################################################################
	#
	$hash_str = $bin_number . $merchant_id . $merchant_salt;
	$paytr_token=base64_encode(hash_hmac('sha256', $hash_str, $merchant_key, true));
	$post_vals=array(
		'merchant_id'=>$merchant_id,
		'bin_number'=>$bin_number,
		'paytr_token'=>$paytr_token
	);
	############################################################################################

	$ch=curl_init();
	curl_setopt($ch, CURLOPT_URL, "https://www.paytr.com/odeme/api/bin-detail");
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
	curl_setopt($ch, CURLOPT_POST, 1) ;
	curl_setopt($ch, CURLOPT_POSTFIELDS, $post_vals);
	curl_setopt($ch, CURLOPT_FRESH_CONNECT, true);
	curl_setopt($ch, CURLOPT_TIMEOUT, 20);

	$result = @curl_exec($ch);

	if(curl_errno($ch))
		die("PAYTR BIN detail request timeout. err:".curl_error($ch));

	curl_close($ch);
	
	$result=json_decode($result,1);
		
	if($result['status']=='error')
		die("PAYTR BIN detail request error. Error:".$result['err_msg']);
	elseif($result['status']=='failed')
		die("BIN is not defined. (For example, a foreign card)");
	else
		print_r($result);